pkg/controller/podgroup.go

```diff
@@ -395,6 +405,31 @@ func calculatePriorityClassName(
 	}
 }
 
+// calculatePriority get priority
+// 1. get priority with priorityClass's value
+// 2. get priority with priority
+func calculatePriority(podSpec *corev1.PodSpec, pcLister schedulinglisters.PriorityClassLister) int32 {
+	if podSpec == nil {
+		return 0
+	}
+	var priority int32
+	pcName := podSpec.PriorityClassName
+	if len(pcName) != 0 && pcLister != nil {
+		if priorityClass, err := pcLister.Get(pcName); err != nil {
+			klog.Warningf("Ignore priority class %q: %v", pcName, err)
+			if podSpec.Priority != nil {
+				priority = *podSpec.Priority
```

Comment:
Is this process needed? IIUC, the `priority` field is automatically filled out by the priority controller using `priorityClass` field. And, we can not modify the `priority` field since the operations against the field is rejected by priority admission controller.

Do you have any reason that you want to fill out `priority` field manually?

Reply:
> User can just set priority value with any number in pod spec,

@lowang-bh IIUC, users can not set priority since the operation will be rejected by the priority admission controller.
You can see the behavior the following manifest:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - name: nginx
    image: nginx
    imagePullPolicy: IfNotPresent
  priority: 100
```

Reply:
So if users set priority to podSpec in MPIJob, the mpijob-controller can not start pods since the pods will be rejected by the priority admission controller.

Reply:
OK, I see now. Thanks.